.github/workflows/nix-package-test.yml

```
@@ -0,0 +1,87 @@
+# Workflow to test Nix packaging on pull requests
+name: 🧪 Test Nix Package
+
+on:
+  pull_request:
+    branches: ["main"]
+
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+jobs:
+  test-nix-package:
+    name: Test Nix Package Build
+    runs-on: ubuntu-latest
+    
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v4
+
+    - name: Install Nix
+      uses: cachix/install-nix-action@v31
+      with:
+        extra_nix_config: |
+          experimental-features = nix-command flakes
+  
+    - name: Set up Cachix (read-only for testing)
+      uses: cachix/cachix-action@v16
+      with:
+        name: qgiswebsite
+        # Use public cache for reading, no auth token needed
+
+    - name: Run flake checks
+      run: nix flake check --show-trace
```

Otherwise, binary cache wouldn't be used:

```suggestion
      run: nix flake check --show-trace --accept-flake-config
```